### PR TITLE
#1481 Makes %<cluc> work consistently, and at all

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/42/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/55/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1737/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/40/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1632/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/166/

File: proxy/logging/LogAccessHttp.cc [hunk 1]
```diff
@@ -71,7 +71,7 @@ LogAccessHttp::LogAccessHttp(HttpSM *sm)
     m_proxy_resp_content_type_str(nullptr),
     m_proxy_resp_content_type_len(0),
     m_cache_lookup_url_canon_str(nullptr),
-    m_cache_lookup_url_canon_len(0)
+    m_cache_lookup_url_canon_len(-1)
```
Comment (bryancall) [inline proxy/logging/LogAccessHttp.cc]: Should have consistency on the len initialization values.  Set them all to -1?

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/298/

File: proxy/logging/LogAccessHttp.cc (hunk 1 above)
Comment (zwoop) [inline proxy/logging/LogAccessHttp.cc]: Maybe, not sure which (if any) should stay zero though. But it seems the URL ones ought to be consistently -1. Let me check that.

Comment (zwoop) [inline proxy/logging/LogAccessHttp.cc]:
```
diff --git a/proxy/logging/LogAccessHttp.cc b/proxy/logging/LogAccessHttp.cc
index 95a620e..b3b7490 100644
--- a/proxy/logging/LogAccessHttp.cc
+++ b/proxy/logging/LogAccessHttp.cc
@@ -57,9 +57,9 @@ LogAccessHttp::LogAccessHttp(HttpSM *sm)
     m_server_response(nullptr),
     m_cache_response(nullptr),
     m_client_req_url_str(nullptr),
-    m_client_req_url_len(0),
+    m_client_req_url_len(-1),
     m_client_req_url_canon_str(nullptr),
-    m_client_req_url_canon_len(0),
+    m_client_req_url_canon_len(-1),
     m_client_req_unmapped_url_canon_str(nullptr),
     m_client_req_unmapped_url_canon_len(-1),
     m_client_req_unmapped_url_path_str(nullptr),
@@ -67,9 +67,9 @@ LogAccessHttp::LogAccessHttp(HttpSM *sm)
     m_client_req_unmapped_url_host_str(nullptr),
     m_client_req_unmapped_url_host_len(-1),
     m_client_req_url_path_str(nullptr),
-    m_client_req_url_path_len(0),
+    m_client_req_url_path_len(-1),
     m_proxy_resp_content_type_str(nullptr),
-    m_proxy_resp_content_type_len(0),
+    m_proxy_resp_content_type_len(-1),
     m_cache_lookup_url_canon_str(nullptr),
     m_cache_lookup_url_canon_len(-1)
 {
```

But, I think I'd prefer to defer this until after we land this, to minimize last minute breakage of 7.1.0.

Comment (bryancall) [inline proxy/logging/LogAccessHttp.cc]: Yeah, it would be better to do it in another PR
